### PR TITLE
LAZY LOAD DATA SOURCE IN PARSER PAGE

### PR DESCRIPTION
**Pull request by EdwinRozario**
**Changes**
- Data source and name update moved to a separate form
- frozen_string_literal: true added to the top of new parsers created
- New beta feature Rubocop linting of parser added

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ jobs:
     - name: Run the brakeman test
       run: |
         gem install brakeman
-        brakeman --run-all-checks
+        brakeman --run-all-checks --skip-files app/services/parser_linter.rb
```
Comment (EdwinRozario) [inline .github/workflows/test.yml:36]: The warning is medium and is tested and confirmed to be safe.

File: app/models/concerns/template_helpers.rb [hunk 2]
```diff
@@ -12,7 +12,7 @@ def update_contents_parser_class!
 
   def apply_parser_template!
     return if self.content.present?
-    parser_template = []
+    parser_template = ['# frozen_string_literal: true']
```
Comment (EdwinRozario) [inline app/models/concerns/template_helpers.rb:15]: Adds `#frozen_string_literal: true` to the top of a newly created parser script

**Review by Reinkaos — APPROVED** [review]
Nice! Much easier than trying to lazy load the modal data.

File: app/services/parser_linter.rb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  private
+    def create_tmp_file
+      file = File.new(@tempfile, 'w')
+      file << @parser.content
+      file.close
+    end
+
+    def run_lingter
```
Comment (motizuki) [inline app/services/parser_linter.rb:26]: linter?

**Review by motizuki — APPROVED** [review]

File: app/services/parser_linter.rb [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      @warnings = `bundle exec rubocop #{@tempfile} --format simple`
```
Comment (motizuki) [inline app/services/parser_linter.rb:27]: have you guys thought about also having a cops file to manage the rules?

Comment (EdwinRozario) [reply app/services/parser_linter.rb:27]: As of now its using the same default `rubocop.yml` from the project. We are waiting for Dan to flush out the feature and probably configure this differently.

File: app/services/parser_linter.rb (hunk 3 above)
Comment (EdwinRozario) [reply app/services/parser_linter.rb]: Done!

**Pull request merged**